Source/BasicVideoEffect.h

```diff
@@ -130,38 +52,37 @@ namespace FFmpegInteropX
                 t + b,			t + b,			t + b,			0
             };
 
-            auto colorMatrixEffect = ref new ColorMatrixEffect();
-            colorMatrixEffect->ColorMatrix = matrix;
-            colorMatrixEffect->Source = source;
+            auto colorMatrixEffect = winrt::Microsoft::Graphics::Canvas::Effects::ColorMatrixEffect();
```

Comment:
I avoided using namespace directives cause in some cases it would cause ambiguity problems.

Reply:
The only place where we seem need them is when accessing FFmpegInteropX classes from within FFmpegInteropX::implementation class. For all other namespaces, usings work fine. At least, I just cleaned up namespaces on the FFmpegMediaSource and saw no issues with that (except as I said, for direct FFmpegInteropX type access).

Reply:
We also use some WRL headers and that causes problems because both c++/winRT and WRL define IInspectable.

Reply:
There are a few cases where collisions occur. IUnknown is one other interface that is defined twice. But that should not lead us to not have namespace usings at all. That makes code very verbose. It is always possible to fix collisions by using explicit namespaces where required.

Reply:
Well, I guess we can work out backwards. But using any single using namespace winrt::blablabla includes IInspectable and IUnknown. But I don't think we should do it now.

Reply:
It's not a problem. Just add explicit namespace where you use IInspectable or IUnknown. Then there is no collision. I will cleanup some of the bigger classes before merging this.

Reply:
I actually tried that approach and got tangled in the spaghetiness of microsoft header libraries. Eventually I gave up and simply fully qualified almost everywhere. 

It might be that you're right and maybe my errors were actually ghost errors that the C++ compiler likes to produce, and now that we have a clean state we can work back and remove some of the full qualifiers. 

We should also consider removing the old "ffmpeginteropx" namespace that contains the pure c++ classes, and leave those outside a namespace. This standalone namespace sometimes conflicts with winrt::fffmpeginteropx

Reply:
I also thought about removing the namespace. Even the ::implementation classes can be removed from namespace. It does not matter where they are or how they are called. They just need to be there with the right template parameters. Microsoft does not use any namespaces for their implementation classes in WinUI 3.

Reply:
I agree with removing the ::implementation namespace if it helps. That would greatly simplify our casts between winRT and implementation types.

Reply:
I already fixed up the bigger classes now. Removing ::implementation is a bigger task. I think we can do that later (unless you like to do it now).

Reply:
I can give it a try during the weekend. Unless you want to merge this now.

Reply:
I am not in a hurry. Depends on how fast the other issues are closed. I definitely want to take a closer look at the event handler thing. C++/WinRT is different than C++/CX in that area, if I understood correctly. I want to be sure that we get the lifetime things correct.